### ISSUE TITLE
Some CSS properties and types do not have all values

### ISSUE THREAD

**Issue by queengooborg** [w3c/webref#1181]
I have been working on a big update to the [mdn-bcd-collector](https://github.com/openwebdocs/mdn-bcd-collector) project to parse values directly out of enumerated types, so that the collector is able to better track new CSS property values when specifications update.

However, I've noticed that there are a number of properties and types that do not define all of the applicable values:

- `display` property (`css-display` spec)
  - Does not have `values` entries for any of the enumerated types (ex. `<display-outside>`)
- `<layout-box>`, `<paint-box>`, `<coord-box>` (`css-box` spec)
  - No `values` entries for the enumerated types they build on
- `<counter-style>` (`css-counter-styles` spec)
  - No `values` entry for `<counter-style-name>`

For now, we're working around the problem with a manually curated list, but it'd be helpful if these type values were included upstream!

**Comment by tidoust** [comment]
I'm afraid that you're going to have a hard time relying on the `values` entry of properties in CSS extracts. Webref merely tries to capture what specs define. The `values` entry is not the result of parsing the value of a property to extract a list of enumerated types. It merely lists whatever the underlying spec defines through a `"value"` definition for the property.

Depending on the spec, this may be:
1. Actual values of an enumerated types. Exactly what you need. But that's not the only possibility and there is no guarantee that the spec will define any/all values.
2. A specific value that the spec wants to higlight, for instance the `css-display` spec only defines `list-item` for the `display` property, which is one of the possibilities of `<display-listitem>`. It does not define `<display-listitem>` or any other value though. The enumerated types are merely seen as intermediary grammar constructs.
3. A more complex grammar construct, such as the one you mention for the `font-size-adjust` property.

Ideally, the specs would be more explicit about values that a property can take. In some cases, the fix is easy because the spec already goes through the values and all that is needed is a bit of extra markup. In many cases though, more editorial work is needed. We haven't tried to push in that direction yet. That would be the right "upstream" ;)

In short, the CSS extracts only *appear to contain* what you're looking for, but the `values` entry of a property merely lists the values that the spec defines, not the values that the property may actually take.

**Comment by tidoust** [comment]
I note that the new [major version of `@webref/css`](https://www.npmjs.com/package/@webref/css/v/7.0.1-alpha) no longer contains the `values` keys for properties. They could be re-introduced if needed, but it seems to me that we should switch to a solution based on parsing the syntax of the property to extract possible keyword values instead of relying on the spec to define the values. At a minimum, if we rely on the specs, we should have a way to detect when values are missing.

Side note: main breaking change in the new version is that the package now contains a consolidated version of CSS extracts. See [changelog](https://github.com/w3c/webref/blob/main/packages/css/CHANGELOG.md#v700-alpha---2025-06-10) for instructions on how to upgrade.
